### PR TITLE
Matt Bricker Reunion Dude

### PR DESCRIPTION
**Pull request by brickstar**
Ran out of time - should have allowed more time, also watching a 3yo while doing this wasn't ideal.
Have some spec questions, outlined in comments.
Commit messages also present some questions in Iteration4.
Thanks!

### CODE REVIEW

File: lib/activity.rb [hunk 1]
```diff
@@ -1,2 +1,27 @@
 class Activity
+  attr_reader :activity_name, :participants
+  def initialize(name)
+    @activity_name = name
+    @participants = Hash.new(0)
```
Comment (mikedao) [inline lib/activity.rb]: Why the default value here?

File: lib/activity.rb [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+  def total_cost
+    @participants.values.sum
+  end
+
+  def split_total_cost
+    total_cost / @participants.length
+  end
+
+  def calculate_payment_offset
+    offset = {}
```
Comment (mikedao) [inline lib/activity.rb]: Every time you iterate over something and modify something in the sibling scope, its a sign that a better enumerable can be used.

File: lib/reunion.rb [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+    @location = location
+    @activities = []
+    @participants = [] # was planning on using this if time had allowed
+  end
+
+  def add_activity(activity)
+    @activities << activity
+  end
+
+  def total_activities_cost
```
Comment (mikedao) [inline lib/reunion.rb]: Great

File: lib/reunion.rb [hunk 4]
```diff
@@ -1,2 +1,28 @@
...
+    @activities.map do |activity|
+      activity.total_cost
+    end.sum
+  end
+
+  def reunion_payment_offset
+    offset = {}
```
Comment (mikedao) [inline lib/reunion.rb]: same comment as previous about modifying another variable.

File: lib/activity.rb (hunk 1 above)
Comment (brickstar) [reply lib/activity.rb]: thought the default value would make it dynamic for participants who contribute 0 to an activity

**Pull request closed without merging**